packages/hdwallet-xdefi/src/xdefi.ts

```diff
@@ -63,6 +64,13 @@ export class XDEFIHDWalletInfo implements core.HDWalletInfo, core.ETHWalletInfo
     return chainId === 1;
   }
 
+  public async ethGetChainId(): Promise<number> {
+    const provider: any = await detectEthereumProvider({ mustBeMetaMask: false, silent: false, timeout: 3000 });
+    // chainId as hex string
+    const chainId: string = await provider.request({ method: "eth_chainId" });
+    return parseInt(chainId, 16);
+  }
+
```

Comment:
I think we only need this method in the `XDEFIHDWallet` class.

Reply:
Seems to work without, removed it in a53319d3ad68006635a6bba6dd5d8079c718d32d